osm2streets/src/road.rs

```diff
@@ -467,6 +515,23 @@ impl Road {
         Ok((left, right))
     }
 
+    pub fn intersection_at(&self, end: RoadEnd) -> IntersectionID {
```

Comment:
These new helpers are nice; there's probably many places throughout the code where they can come in handy!

Reply:
Using `RoadEnd` has been indispensable while dealing with the possibility space explosion in `crate::movements`, and these helpers remove a lot of noise from those implementation sites that make it possible to see the (still very complicated) logic there.